File: Q34.py
#Q34.py
class Solution:
    def searchRange(self, a, t):
        def lb(x):
            l=0; r=len(a)
            while l<r:
                m=(l+r)//2
                if a[m]<x: l=m+1
                else: r=m
            return l
        L=lb(t)
        if L==len(a) or a[L]!=t: return [-1,-1]
        R=lb(t+1)-1
        return [L,R]

File: Q34.cpp
//Q34.cpp
class Solution {
public:
    int lb(vector<int>& a,int t){
        int l=0,r=a.size();
        while(l<r){
            int m=(l+r)>>1;
            if(a[m]<t) l=m+1;
            else r=m;
        }
        return l;
    }
    vector<int> searchRange(vector<int>& a, int t) {
        int L=lb(a,t);
        if(L==a.size() || a[L]!=t) return {-1,-1};
        int R=lb(a,t+1)-1;
        return {L,R};
    }
};

